src/dataframe/dataframe.jl

```diff
@@ -1026,42 +1026,30 @@ function hcat!(df1::DataFrame, df2::AbstractDataFrame;
     return df1
 end
 
-# definition required to avoid hcat! ambiguity
-hcat!(df1::DataFrame, df2::DataFrame;
-      makeunique::Bool=false, copycols::Bool=true) =
-    invoke(hcat!, Tuple{DataFrame, AbstractDataFrame}, df1, df2,
-           makeunique=makeunique, copycols=copycols)::DataFrame
-
-hcat!(df::DataFrame, x::AbstractVector; makeunique::Bool=false, copycols::Bool=true) =
-    hcat!(df, DataFrame(AbstractVector[x], [:x1], copycols=copycols),
-          makeunique=makeunique, copycols=copycols)
-hcat!(x::AbstractVector, df::DataFrame; makeunique::Bool=false, copycols::Bool=true) =
-    hcat!(DataFrame(AbstractVector[x], [:x1], copycols=copycols), df,
-          makeunique=makeunique, copycols=copycols)
-hcat!(x, df::DataFrame; makeunique::Bool=false, copycols::Bool=true) =
-    throw(ArgumentError("x must be AbstractVector or AbstractDataFrame"))
-hcat!(df::DataFrame, x; makeunique::Bool=false, copycols::Bool=true) =
-    throw(ArgumentError("x must be AbstractVector or AbstractDataFrame"))
+# TODO: after deprecation remove AbstractVector methods
```

Comment:
It will be easier to keep them here, as I will have to change the method in line https://github.com/JuliaData/DataFrames.jl/pull/2777/files#diff-151ff71ca0da68d81f47f5ffffb30d032f8b63682ac14ba5c81305b4836dd3a3R1047 here anyway + most likely we will add the methods handling Tables.jl tables.

In conclusion: I think it will be less error prone to keep all the code that needs to be changed in one place.